server/src/internalClusterTest/java/org/elasticsearch/action/termvectors/MultiTermVectorsIT.java

```diff
@@ -34,27 +35,33 @@ public void testDuelESLucene() throws Exception {
         // we generate as many docs as many shards we have
         TestDoc[] testDocs = generateTestDocs("test", testFieldSettings);
 
-        DirectoryReader directoryReader = indexDocsWithLucene(testDocs);
-        AbstractTermVectorsTestCase.TestConfig[] testConfigs = generateTestConfigs(20, testDocs, testFieldSettings);
+        DirectoryReader directoryReader = null;
+        try {
+            directoryReader = indexDocsWithLucene(testDocs);
 
-        MultiTermVectorsRequestBuilder requestBuilder = client().prepareMultiTermVectors();
-        for (AbstractTermVectorsTestCase.TestConfig test : testConfigs) {
-            requestBuilder.add(getRequestForConfig(test).request());
-        }
+            AbstractTermVectorsTestCase.TestConfig[] testConfigs = generateTestConfigs(20, testDocs, testFieldSettings);
 
-        MultiTermVectorsItemResponse[] responseItems = requestBuilder.get().getResponses();
+            MultiTermVectorsRequestBuilder requestBuilder = client().prepareMultiTermVectors();
+            for (AbstractTermVectorsTestCase.TestConfig test : testConfigs) {
+                requestBuilder.add(getRequestForConfig(test).request());
+            }
 
-        for (int i = 0; i < testConfigs.length; i++) {
-            TestConfig test = testConfigs[i];
-            MultiTermVectorsItemResponse item = responseItems[i];
-            if (test.expectedException != null) {
-                assertTrue(item.isFailed());
-                continue;
-            } else if (item.isFailed()) {
-                fail(item.getFailure().getCause().getMessage());
+            MultiTermVectorsItemResponse[] responseItems = requestBuilder.get().getResponses();
+
+            for (int i = 0; i < testConfigs.length; i++) {
+                TestConfig test = testConfigs[i];
+                MultiTermVectorsItemResponse item = responseItems[i];
+                if (test.expectedException != null) {
+                    assertTrue(item.isFailed());
+                    continue;
+                } else if (item.isFailed()) {
+                    fail(item.getFailure().getCause().getMessage());
+                }
+                Fields luceneTermVectors = getTermVectorsFromLucene(directoryReader, test.doc);
+                validateResponse(item.getResponse(), luceneTermVectors, test);
             }
-            Fields luceneTermVectors = getTermVectorsFromLucene(directoryReader, test.doc);
-            validateResponse(item.getResponse(), luceneTermVectors, test);
+        } finally {
+            IOUtils.close(directoryReader);
```

Comment:
Failing to close the reader can lead to thread leaks if the randomized searcher enables concurrency.